samples/bluetooth/unicast_audio_server/src/main.c

```diff
@@ -230,24 +230,23 @@ static struct bt_audio_capability_ops lc3_ops = {
 
 #if defined(CONFIG_LIBLC3CODEC)
 
-static void stream_recv_lc3_codec(struct bt_audio_stream *stream, struct net_buf *buf)
+static void stream_recv_lc3_codec(struct bt_audio_stream *stream,
+				  const struct bt_iso_recv_info *info,
+				  struct net_buf *buf)
 {
 	uint8_t err;
-	/* TODO: If there is a way to know if the controller supports indicating errors in the
-	 *       payload one could feed that into bad-frame-indicator. The HCI layer allows to
-	 *       include this information, but currently there is no controller support.
-	 *       Here it is assumed that reveiving a zero-length payload means a lost frame -
-	 *       but actually it could just as well indicate a pause in the stream.
-	 */
-	const uint8_t bad_frame_indicator = buf->len == 0 ? 1 : 0;
-	uint8_t *in_buf = (bad_frame_indicator ? NULL : buf->data);
+
+	if (info->flags != BT_ISO_FLAGS_VALID) {
+		printk("Ignoring invalid Audio packet\n");
+		return;
+	}
```

Comment:
Actually what we need to do in this case is to request the codec to generate Packet loss concealment data. You do that by calling `lc3_decode()` with a `null` pointer - like:
`uint8_t *in_buf = (info->flags != BT_ISO_FLAGS_VALID ) ? NULL : buf->data;`

Reply:
Updated it to be more similar to your original handling